### DISCUSSION TITLE
in-person payments and payment_method_types with stripe

### DISCUSSION THREAD

**Discussion by tyler71** [medusajs/medusa#4575] [Q&A] [answered]
I'm hoping to accept both in-person and e-commerce purchases with medusajs / stripe.
in-person to be completed with the [Stripe Reader M2](https://stripe.com/terminal/stripe-reader)

For this to work, I need a payment intent with `payment_method_types` to be `["card_present", "card"]`

```python
# python example
stripe.PaymentIntent.create(
  currency="usd",
  payment_method_types=["card", "card_present"],
  capture_method="automatic",
  amount=1000,
)
```

Looking through the [source code](https://github.com/medusajs/medusa/blob/develop/packages/medusa-payment-stripe/src/core/stripe-base.ts#L43C1-L60C4), it seems if I set the environment variable `payment_method_types` I can achieve this.
Can anyone confirm, or offer a better way of handling this?
```typescript
  getPaymentIntentOptions(): PaymentIntentOptions {
    const options: PaymentIntentOptions = {}

    if (this?.paymentIntentOptions?.capture_method) {
      options.capture_method = this.paymentIntentOptions.capture_method
    }

    if (this?.paymentIntentOptions?.setup_future_usage) {
      options.setup_future_usage = this.paymentIntentOptions.setup_future_usage
    }

    if (this?.paymentIntentOptions?.payment_method_types) {
      options.payment_method_types =
        this.paymentIntentOptions.payment_method_types
    }

    return options
  }
```

**Comment by srindom** [comment]
I know that @josipmatichr has experience with accepting Stripe Terminal payments. Maybe he can share some learnings?

**Comment by josipmatichr** [comment, marked as answer]
@srindom of course! @tyler71 I think you won't be able to set `payment_method_types` through environment variables since `paymentIntentOptions` should be defined through child class. I would suggest [creating](https://docs.medusajs.com/development/services/create-service) a new service that will extend base stripe service, something like this should work: 

```
import StripeBase from "medusa-payment-stripe/dist/core/stripe-base"
import { PaymentIntentOptions } from "medusa-payment-stripe"

class StripeTerminalProviderService extends StripeBase {
  static identifier = "stripe-terminal"

  constructor(_, options) {
    super(_, options)
  }

  get paymentIntentOptions(): PaymentIntentOptions {
    return {
      payment_method_types: ["card", "card_present"],
      capture_method: "automatic",
    }
  }
}

export default StripeTerminalProviderService
```

Just remember to enable "stripe-terminal" in the region payment provider settings :)

Hope that helps!

**Comment by tyler71** [comment]
Thank you! This is very helpful
